File: beziercurve/src/main/java/com/apt/beziercurve/MainActivity.java
package com.apt.beziercurve;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.waveView).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                WaveView waveView = (WaveView) v;
                waveView.startAnim();
            }
        });
    }
}

File: reminddemo/src/main/java/com/zff/reminddemo/MainActivity.kt
package com.zff.reminddemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.FrameLayout
import android.widget.Toast
import kotlinx.android.synthetic.*
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn.post {
            val remindView:RemindView = RemindView(this)
            remindView.setOnClickListener {
                Toast.makeText(this,"我在点击范围内",Toast.LENGTH_SHORT).show()
            }
            remindView.setAnchorView(btn)
            (window.decorView as FrameLayout).addView(remindView)
        }
    }
}
